python/tank/platform/engine.py

```diff
@@ -1052,7 +1054,14 @@ def register_command(self, name, callback, properties=None):
         # to highlight this state. This is used by the tank_command
         # execution logic to correctly dispatch the callback during
         # runtime.
-        arg_spec = inspect.getargspec(callback)
+        # getargspec has been deprecated in Python 3 and generates a copious
+        # amount of warnings, so use getfullargspec which is backwards
+        # compatible in Python 3. Unfortunately, it doesn't exist in Python
+        # 2 and six doesn't offer a wrapper for it.
+        if six.PY2:
+            arg_spec = inspect.getargspec(callback)
+        else:
+            arg_spec = inspect.getfullargspec(callback)
```

Comment:
I think we do this only in one or two places. If it becomes more prevalent it makes sense to refactor it, but let's not add stuff to sgsix because we use it once or twice.